src/content/docs/es/guides/actions.mdx

```diff
@@ -0,0 +1,702 @@
+
+Para autorizar solicitudes de acciones, agrega una verificación de autenticación a tu controlador de acciones. Es posible que desees usar [una biblioteca de autenticación](/es/guides/authentication/) para manejar la gestión de sesiones y la información del usuario.
+
+Las acciones exponen el objeto `APIContext` completo para acceder a las propiedades pasadas desde el middleware utilizando `context.locals`. Cuando un usuario no está autorizado, puedes generar un `ActionError` con el código `UNAUTHORIZED`:
+
+```ts title="src/actions/index.ts" {6-8}
+import { defineAction, ActionError } from 'astro:actions';
+
+export const server = {
+  getUserSettings: defineAction({
+    handler: async (_input, context) => {
+      if (!context.locals.user) {
+        throw new ActionError({ code: 'UNAUTHORIZED' });
+      }
+      return { /* datos en caso de éxito */ };
+    }
+  })
+}
+```
+
+### Acciones de puerta desde middleware
+
+<p><Since v="5.0.0" /></p>
+
+Astro recomienda autorizar las sesiones de usuario desde el controlador de tu acción para respetar los niveles de permiso y la limitación de velocidad por acción. Sin embargo, también puedes bloquear las solicitudes a todas las acciones (o a un subconjunto de acciones) desde el middleware.
+
+Usa la función `getActionContext()` desde tu middleware para recuperar información sobre las solicitudes de acciones entrantes. Esto incluye el nombre de la acción y si esa acción se llamó usando una función de llamada a procedimiento remoto (RPC) del lado del cliente (por ejemplo, `actions.blog.like()`) o un formulario HTML.
+
+El siguiente ejemplo rechaza todas las solicitudes de acciones que no tienen un token de sesión válido. Si falla la verificación, se devuelve una respuesta "Prohibido". Nota: este método garantiza que las acciones solo sean accesibles cuando hay una sesión presente, pero **no** sustituye la autorización segura.
+
+```ts title="src/middleware.ts"
+import { defineMiddleware } from 'astro:middleware';
+import { getActionContext } from 'astro:actions';
+
+export const onRequest = defineMiddleware(async (context, next) => {
+  const { action } = getActionContext(context);
+  // Verifica si se llamó a la acción desde una función del lado del cliente
+  if (action?.calledFrom === 'rpc') {
+    // Si es así, busca un token de sesión de usuario
+    if (!context.cookies.has('user-session')) {
+      return new Response('Forbidden', { status: 403 });
+    }
+  }
+
+  context.cookies.set('user-session', /* token de sesión */);
+  return next();
+});
+```
+
+## Llamar a acciones desde componentes Astro y puntos de conexión del servidor
+
+Puedes llamar a acciones directamente desde los scripts de componentes Astro usando el contenedor `Astro.callAction()` (o `context.callAction()` cuando usas un [punto de conexión del servidor](/es/guides/endpoints/#puntos-de-conexion-de-servidor-rutas-api)). Esto es común para reutilizar la lógica de tus acciones en otro código de servidor.
```

Comment:
Generally translations, and the other translated Spanish pages don't translate "endpoint" and keep it as is, I would appreciate if you could update this page to reflect that so we keep it consistent.